packages/drivers/routerlicious-driver/src/documentStorageService.ts

```diff
@@ -183,21 +188,18 @@ export class DocumentStorageService implements IDocumentStorageService {
         /** Previous snapshot, subtree relative to this path part */
         previousSnapshot: ISnapshotTree,
     ): string {
+        assert(path.length > 0, "Expected at least 1 path part");
         const key = path[0];
         if (path.length === 1) {
```

Comment:
this IF should be removed, right?

Reply:
Why? I think it should stay.

We recursively call this function until we are in the innermost path part: `path.length === 1`, which breaks the recursion and returns the ID instead (and requires split based on blob vs tree).

The fix prevents starting at zero path lengths and hitting an infinite loop.